edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/MetricsSerializer.cs

```diff
@@ -22,13 +23,13 @@ public static IEnumerable<byte> MetricsToBytes(IEnumerable<Metric> metrics)
                 x.Select(m => new RawMetricValue(m.TimeGeneratedUtc, m.Value))));
         }
 
-        static IEnumerable<byte> MetricGroupsToBytes(string name, string tags, IEnumerable<RawMetricValue> rawValues)
+        static IEnumerable<byte> MetricGroupsToBytes(string name, Dictionary<string, string> tags, IEnumerable<RawMetricValue> rawValues)
```

Comment:
```suggestion
        static IEnumerable<byte> MetricGroupsToBytes(string name, IReadonlyDictionary<string, string> tags, IEnumerable<RawMetricValue> rawValues)
```

Reply:
This is intentionally a readonly dictionary as Lee and I decided to allow for custom tags to be added in the case where the scraper is used elsewhere. For example in the metrics collector I edit the tags to add a guid.

Reply:
Philip and I discussed and we will make the Metric's dictionary readonly. In the metrics collector I will have a custom metric class with custom tags field.